### PR TITLE
feat(load): modify load check to check if storage radius is bigger then max

### PR DESCRIPTION
**Pull request by gacevicljubisa**
### Description

This PR addresses [issue #400](https://github.com/ethersphere/beekeeper/issues/400). It adds a check in the upload worker to stop uploads if the storage radius exceeds a configured maximum value.

### Purpose

To ensure uploads only proceed when the storage radius is within acceptable limits.

### Implementation

- Checks storage radius from `/reservestate` or `/status` before uploading.
- If the radius is too high, the worker sleeps for 5-10 minutes before retrying.
- Applies to the smoke/load job, not the smoke job.
- Introduces a new config option for the max allowed radius.

### Testing

I need reviewers to verify the flow and suggest improvements. Do we need a new config option for the sleep time, or can we utilize an existing one? Currently, three requests to `/reservestate` are made before uploading: one that is added and two in `CreatePostageBatch` (one prior to buying the batch and one after buying the batch). Please provide the correct parameters for the load check to reproduce the behavior.

### CODE REVIEW

File: pkg/check/smoke/load.go [hunk 1]
```diff
@@ -102,24 +112,46 @@ func (c *LoadCheck) Run(ctx context.Context, cluster orchestration.Cluster, opts
...
+						if rs.StorageRadius <= o.MaxStorageRadius {
+							break
+						}
+						c.log.Infof("waiting for StorageRadius to decrease. Current: %d, Max: %d", rs.StorageRadius, o.MaxStorageRadius)
+
+						select {
+						case <-ctx.Done():
+							c.log.Infof("context done in StorageRadius check: %v", ctx.Err())
+							return
+						case <-time.After(o.TxOnErrWait): // TODO: Adjust the sleep duration as needed
```
Comment (martinconic) [inline pkg/check/smoke/load.go:142]: What should be the adjustment here?

Comment (gacevicljubisa) [reply pkg/check/smoke/load.go:142]: Regarding context check, if global timeout happens, the context should be canceled, and we should stop execution.

Regarding the note: 
> **TODO:** Adjust the sleep duration as needed

I have already put a question in the PR description:

> Do we need a new config option for the sleep time, or can we utilize an existing one?

File: pkg/check/smoke/load.go [hunk 2]
```diff
@@ -102,24 +112,46 @@ func (c *LoadCheck) Run(ctx context.Context, cluster orchestration.Cluster, opts
...
 					}
 
+					// Check the StorageRadius
+					for {
+						rs, err := clients[txName].ReserveState(ctx)
+						if err != nil {
+							c.log.Infof("error getting state: %v", err)
+							return
+						}
+						if rs.StorageRadius <= o.MaxStorageRadius {
```
Comment (istae) [inline pkg/check/smoke/load.go:133]: should be strictly `<`

File: pkg/check/smoke/load.go (hunk 1 above)
Comment (istae) [reply pkg/check/smoke/load.go:142]: this check does not need to happen so often, something like every 5 min, at the max.
let's add a new config option for this wait, with default being 5 min.

**Review by istae — CHANGES_REQUESTED** [review]

**Review by istae — APPROVED** [review]

**Pull request merged**